File: app/src/main/java/com/riders/thelab/ui/youtubelike/YoutubeListClickListener.java
package com.riders.thelab.ui.youtubelike;

import androidx.annotation.NonNull;

import com.google.android.material.imageview.ShapeableImageView;
import com.google.android.material.textview.MaterialTextView;
import com.riders.thelab.data.local.model.Video;


public interface YoutubeListClickListener {
    void onYoutubeItemClicked(
            @NonNull ShapeableImageView thumbShapeableImageView,
            @NonNull MaterialTextView titleTextView,
            @NonNull MaterialTextView descriptionTextView,
            Video video,
            int position);
}

File: app/src/main/java/com/riders/thelab/ui/youtubelike/YoutubeListClickListener.kt
package com.riders.thelab.ui.youtubelike

import android.view.View
import com.google.android.material.imageview.ShapeableImageView
import com.google.android.material.textview.MaterialTextView
import com.riders.thelab.core.data.local.model.Video

interface YoutubeListClickListener {

    fun onYoutubeItemClicked(view: View, video: Video)

    fun onYoutubeItemClicked(
        thumbShapeableImageView: ShapeableImageView,
        titleTextView: MaterialTextView,
        descriptionTextView: MaterialTextView,
        video: Video,
        position: Int
    )
}